packages/hoprd/crates/hoprd-misc/src/lib.rs

```diff
@@ -18,14 +17,19 @@ struct PackageJsonFile {
     version: String
 }
 
+/// Helper function to convert string-convertible types (like errors) to JsValue
+fn as_jsvalue<T>(v: T) -> JsValue where T: Display {
+    JsValue::from(v.to_string())
+}
+
 /// Reads the package.json file of hoprd and determines it's version.
 #[wasm_bindgen]
-pub fn get_hoprd_version(package_file: &str) -> Result<String, JsValue> {
+pub fn get_package_version(package_file: &str) -> Result<String, JsValue> {
 
-    let file_data = real::read_file(package_file);
+    let file_data = Vec::from(real::read_file(package_file)?);
```

Comment:
If I'm not entirely wrong, you can also use `Box` since the array does not need to be resizable